hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -782,7 +788,7 @@ observeCloseTx utxo tx = do
                 , closedContestationDeadline = closeContestationDeadline
                 }
           , headId
-          , snapshotNumber
+          , snapshotNumber = UnsafeSnapshotNumber sn
```

Comment:
Defining a function `SnapshotNumber.fromChain` or so could make it clear that this is a fine conversion?

Reply:
I've named it `fromChainSnapshot` since we have `fromChain` already in `ContestationPeriod`